pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile

```diff
@@ -0,0 +1,199 @@
+        buildDescription "Env: ${namespace.getWorkspaceName()}\nBranch: ${params.MODULE_BRANCH}\n" + "Config: ${params.CONFIG_TYPE}\n${currentBuild.getBuildCauses()[0].shortDescription}"
+      }
+
+      stage('[Git] Checkout module source') {
+        if(skipStage) {
+          logger.info("Skipping stage")
+          return
+        }
+
+        checkout([
+          $class           : 'GitSCM',
+          branches         : [[name: "*/${featureBranch}"]],
+          extensions       : [[$class: 'RelativeTargetDirectory', relativeTargetDir: module.name],
+                              [$class: 'CleanBeforeCheckout', deleteUntrackedNestedRepositories: true],
+                              [$class: 'AuthorInChangelog'],
+                              [$class: 'SubmoduleOption', recursiveSubmodules: true]],
+          userRemoteConfigs: [[url: "https://github.com/folio-org/${module.name}.git"]]
+        ])
+      }
+
+      stage('[Maven] Compile') {
+        if(skipStage) {
+          logger.info("Skipping stage")
+          return
+        }
+
+        dir(module.getName()) {
+          String origVersion = readMavenPom(file: 'pom.xml').version
+          module.version = "${origVersion}.${featureHash}"
+
+          withMaven(
+            jdk: "${common.selectJavaBasedOnAgent(params.AGENT)}".toString(),
+            maven: Constants.MAVEN_TOOL_NAME,
+            traceability: false,
+            options: [artifactsPublisher(disabled: true)])
+          {
+            sh """
+              mvn versions:set -DnewVersion=${module.getVersion()}
+              mvn package ${mavenArguments}
+            """.stripIndent()
+          }
+
+          String descriptorPath = "target/ModuleDescriptor.json"
+          if (fileExists(descriptorPath)) {
+            module.descriptor = [readJSON(file: descriptorPath)]
+            sh(script: "curl ${Constants.EUREKA_REGISTRY_URL}${module.name}-${module.version} --upload-file ${descriptorPath}")
+          }
+        }
+      }
+
+      stage('[Docker] Build and push') {
+        if(skipStage) {
+          logger.info("Skipping stage")
+          return
+        }
+
+        common.checkEcrRepoExistence(module.getName())
+        docker.withRegistry("https://${Constants.ECR_FOLIO_REPOSITORY}", "ecr:${Constants.AWS_REGION}:${Constants.ECR_FOLIO_REPOSITORY_CREDENTIALS_ID}") {
+          dir(module.getName()) {
+            def image = docker.build("${module.getName()}:${module.getVersion()}", '--no-cache=true --pull=true .')
+            image.push()
+          }
+        }
+      }
+
+      stage('Generate context') {
+        namespace.tenants = eureka.getExistedTenantsFlow()
+        logger.debug("Configured Tenants: ${namespace.tenants}")
+
+        namespace.withApplications(eureka.getEnabledApplicationsFlow(namespace.tenants))
+        logger.debug("Configured Applications: ${namespace.applications}")
+
+        // TODO: debug code to be removed soon
```

Comment:
Please do not forget to remove this one

Reply:
Thanks. 
Thats why there is such comment with TODO.